functions/Find-DbaUserObject.ps1

```diff
@@ -37,292 +37,294 @@ You should have received a copy of the GNU General Public License
-                        ObjectName = $sch.Name
-                        ObjectDetails = $db.Name
-                        }
-                        $out
-                }
-
-                ## database roles
-                if (-not $Pattern)
-                {
-                    $roles = $db.Roles | Where-Object {$_.IsSystemObject -eq 0 -and $_.Owner -ne "dbo"}
-                }
-                else
-                {
-                    $roles = $db.Roles | Where-Object {$_.IsSystemObject -eq 0 -and $_.Owner -eq $Pattern}
-                }
-                foreach ($role in $roles)
-                {
-                    $out = [PSCustomObject]@{
-                        ComputerName = $server.NetName
-                        SqlInstance = $server.InstanceName
-		                ObjectType = "Database Role" 
-                        ObjectOwner = $role.Owner
-                        ObjectName = $role.Name
-                        ObjectDetails = $db.Name
-                        }
-                        $out
-                }
-
-                ## assembly
-                if (-not $Pattern)
-                {             
-                    $Assemblies = $db.Assemblies | Where-Object {$_.IsSystemObject -eq 0 -and $_.Owner -ne "dbo"}
-                }
-                else
-                {
-                    $Assemblies = $db.Assemblies | Where-Object {$_.IsSystemObject -eq 0 -and $_.Owner -eq $Pattern}
-                }
-                foreach ($Assemblie in $Assemblies)
-                {
-                    $out = [PSCustomObject]@{
-                        ComputerName = $server.NetName
-                        SqlInstance = $server.InstanceName
-		                ObjectType = "Database Assembly" 
-                        ObjectOwner = $Assemblie.Owner
-                        ObjectName = $Assemblie.Name
-                        ObjectDetails = $db.Name
-                        }
-                        $out
-                }
-
-              ## synonyms
-                if (-not $Pattern)
-                {             
-                    $Synonymss = $db.Synonyms | Where-Object {$_.IsSystemObject -eq 0 -and $_.Owner -ne "dbo"}
-                }
-                else
-                {
-                    $Synonymss = $db.Synonyms | Where-Object {$_.IsSystemObject -eq 0 -and $_.Owner -eq $Pattern}
-                }
-                foreach ($Synonyms in $Synonymss)
-                {
-                    $out = [PSCustomObject]@{
-                        ComputerName = $server.NetName
-                        SqlInstance = $server.InstanceName
-		                ObjectType = "Database Synonyms" 
-                        ObjectOwner = $Synonyms.Owner
-                        ObjectName = $Synonyms.Name
-                        ObjectDetails = $db.Name
-                        }
-                        $out
-                }
-            }  
-
-        }
-    }
+	process
+	{
+		if ($Pattern -like "*\*")
+		{
```

Comment:
Re: adding a begin - it's true. i was bein lazy 👎 

This was intended to fix people who pass ad\username instead of ad\\username. Do you have the regex that'd only match what i'm looking to fix? I dont want to escape everything but i do want to catch the most common error.